evaluation/ml_bench/run_infer.py

```diff
@@ -114,139 +114,156 @@ def parse_eval_output(output: str) -> Dict[str, float]:
+        except Exception as e:
+            logger.error(f'Error reading evaluation script: {e}')
+            eval_script_content = ''
 
-    try:
-        exit_code, eval_output = sandbox.execute(
-            f'conda run -n {ID2CONDA[instance["github_id"]]} bash {eval_script}'
-        )
-    except Exception as e:
-        logger.error(f'Error running evaluation script: {e}')
-        exit_code = -1
-        eval_output = ''
-
-    if exit_code != 0:
-        logger.warning(f'Evaluation script failed with exit code {exit_code}')
-        logger.warning(f'Output: {eval_output}')
-        metrics = {
-            'success': 'KeyboardInterrupt'
-            in eval_output,  # super-dainiu: assume ``KeyboardInterrupt`` is a success as is done in ML-Bench
-        }
-    else:
-        logger.info(f'Evaluation script succeeded with exit code {exit_code}')
-        logger.info(f'Output: {eval_output}')
-        metrics = {
-            'success': 1,
+        try:
+            exit_code, eval_output = sandbox.execute(
+                f'conda run -n {ID2CONDA[instance["github_id"]]} bash {eval_script}'
+            )
+        except Exception as e:
+            logger.error(f'Error running evaluation script: {e}')
+            exit_code = -1
+            eval_output = ''
+
+        if exit_code != 0:
+            logger.warning(f'Evaluation script failed with exit code {exit_code}')
+            logger.warning(f'Output: {eval_output}')
+            metrics = {
+                'success': 'KeyboardInterrupt'
+                in eval_output,  # super-dainiu: assume ``KeyboardInterrupt`` is a success as is done in ML-Bench
+            }
+        else:
+            logger.info(f'Evaluation script succeeded with exit code {exit_code}')
+            logger.info(f'Output: {eval_output}')
+            metrics = {
+                'success': 1,
+            }
+
+        # Save the output
+        output = {
+            'instance_id': instance['id'],
+            'repo': repo_url,
+            'instruction': instruction,
+            'metadata': metadata,
+            'history': [
+                (event_to_dict(action), event_to_dict(obs))
+                for action, obs in state.history
+            ],
+            'eval_script': eval_script_content,
+            'eval_exit_code': exit_code,
+            'eval_output': eval_output,
+            'metrics': metrics,
         }
 
-    # Save the output
-    output = {
-        'instance_id': instance['id'],
-        'repo': repo_url,
-        'instruction': instruction,
-        'metadata': metadata,
-        'history': [
-            (event_to_dict(action), event_to_dict(obs)) for action, obs in state.history
-        ],
-        'eval_script': eval_script_content,
-        'eval_exit_code': exit_code,
-        'eval_output': eval_output,
-        'metrics': metrics,
-    }
+    except Exception as e:
+        logger.error(f'Error processing instance {instance["id"]}: {e}')
+        output = {
```

Comment:
nit: this statement seems unnecessary, since we are raising the exception afterward